src/container/srv_target.c

```diff
@@ -192,15 +192,15 @@ cont_child_aggregate(struct ds_cont_child *cont, uint64_t *msecs)
 		if (snapshots == NULL)
 			return -DER_NOMEM;
 
-		for (j = 0, k = 0; j < cont->sc_snapshots_nr; j++) {
-			if (cont->sc_snapshots[j] > rebuild_fence) {
+		for (j = 0, insert_idx = 0; j < cont->sc_snapshots_nr; j++) {
```

Comment:
It's better to assert cont->sc_snapshots[j] > cont->sc_snapshots[j - 1] when j > 0.